### PR TITLE
Waiting screen while guessing reconciliation types should be internal to the reconciliation dialog #4877

### LINKED ISSUE #4877: Waiting screen while guessing reconciliation types should be internal to the reconciliation dialog
Currently, when opening the reconciliation dialog, we have the following user experience:
1. user selects the reconciliation service
2. a "Working…" screen appears while OpenRefine guesses reconciliation types. The user cannot do anything while this is shown.
3. the reconciliation configuration dialog appears and the user can start configuring the reconciliation

Sometimes the guessing of reconciliation types actually takes quite some time (depending on the recon endpoint, the internet connection speed, the specifics of the dataset…) and it can be quite frustrating to wait for this process to complete - especially if you already know which type you want to reconcile the data to, or if the service does not use types at all.

### Proposed solution
Instead of having this "Working…" screen, we should directly show the reconciliation configuration dialog. We would have a waiting spinner (similar to the one used in data extension preview) in the sub-area of the dialog which shows guessed types. It could look like this (with an animated spinner):
image

### Alternatives considered

None

### Additional context
Problem observed in a session at Dataharvest 2022 (in a conference room with pretty bad wifi)

### PR DESCRIPTION
**Pull request by ayushrai206**
Fixes #4877 

Changes proposed in this pull request:
- adds the loading message and the spinner to the types container

### CODE REVIEW

**Comment by wetneb** [comment]
Really nice @ayushrai206! It works pretty well! One suggestion I would still have is that the panel on the right, where we can map columns to properties, could be displayed without waiting for the types to be guessed. You can see in this screencast that this panel is initially blank while the types are fetched, and intuitively that should not be needed.
recon

**Review by Abbe98 — CHANGES_REQUESTED** [review]
This looks very good! I only have a minor suggestion.

It would also be nice if it was possible to see and interact with the "columns to properties" column while loading is still happening as suggested by @wetneb.

File: main/webapp/modules/core/scripts/util/dialog.js [hunk 1]
```diff
@@ -147,4 +147,13 @@ DialogSystem.showBusy = function(message) {
     DialogSystem.dismissUntil(level - 1);
   };
 };
+DialogSystem.showBusyReconciling = function(message) {
```
[the remark this answers is not in the document]
Comment (Abbe98) [reply main/webapp/modules/core/scripts/util/dialog.js]: I would also move it from the `DialogSystem` object into `self`.

[the remark this answers is not in the document]
Comment (ayushrai206) [reply main/webapp/modules/core/scripts/util/dialog.js]: @Abbe98 i have also tried to bring back the panel on the right,please have a look

**Review by Abbe98 — CHANGES_REQUESTED** [review]
Functionality vise it appears to work flawlessly. ⭐  Just a minor change request left from my end.

File: main/webapp/modules/core/scripts/reconciliation/standard-service-panel.js [hunk 2]
```diff
@@ -345,3 +346,13 @@ ReconStandardServicePanel.prototype.start = function() {
   );
 };
 
+DialogSystem.showBusyReconciling = function(message) {
```
[the remark this answers is not in the document]
Comment (ayushrai206) [reply main/webapp/modules/core/scripts/reconciliation/standard-service-panel.js]: did it @Abbe98

**Pull request merged**